examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts

```diff
@@ -31,55 +30,96 @@ import {
+  }
+
   const patientAddress = getPatientAddress(answers);
-  patient.address = patientAddress ? [patientAddress] : patient.address;
-  patient.gender = (answers['gender-identity']?.valueCoding?.code as Patient['gender']) || patient.gender;
-  patient.telecom = answers['phone']?.valueString
-    ? [{ system: 'phone', value: answers['phone'].valueString }]
-    : patient.telecom;
-  patient.identifier = answers['ssn']?.valueString
-    ? [
-        {
-          type: {
+  if (patientAddress) {
+    patient.address = [patientAddress];
+  }
+
+  if (answers['gender-identity']?.valueCoding?.code) {
+    patient.gender = answers['gender-identity'].valueCoding.code as Patient['gender'];
+  }
+
+  if (answers['phone']?.valueString) {
+    patient.telecom = [{ system: 'phone', value: answers['phone'].valueString }];
+  }
+
+  if (answers['ssn']?.valueString) {
+    patient.identifier = [
+      {
+        type: {
+          coding: [
+            {
+              system: 'http://terminology.hl7.org/CodeSystem/v2-0203',
+              code: 'SS',
+            },
+          ],
+        },
+        system: 'http://hl7.org/fhir/sid/us-ssn',
+        value: answers['ssn'].valueString,
+      },
+    ];
+  }
+
+  const emergencyContacts = getGroupRepeatedAnswers(questionnaire, response, 'emergency-contact');
+  if (emergencyContacts) {
+    patient.contact = [];
+    for (const contact of emergencyContacts) {
+      patient.contact.push({
+        relationship: [
+          {
             coding: [
               {
-                system: 'http://terminology.hl7.org/CodeSystem/v2-0203',
-                code: 'SS',
+                system: 'http://terminology.hl7.org/CodeSystem/v2-0131',
+                code: 'EP',
+                display: 'Emergency contact person',
               },
             ],
           },
-          system: 'http://hl7.org/fhir/sid/us-ssn',
-          value: answers['ssn'].valueString,
-        },
-      ]
-    : patient.identifier;
+        ],
+        name: getHumanName(contact, 'emergency-contact-'),
+        telecom: [{ system: 'phone', value: contact['emergency-contact-phone']?.valueString }],
+      });
+    }
+  }
 
   setExtension(patient, extensionURLMapping.race, 'valueCoding', answers['race']);
   setExtension(patient, extensionURLMapping.ethnicity, 'valueCoding', answers['ethnicity']);
   setExtension(patient, extensionURLMapping.veteran, 'valueBoolean', answers['veteran-status']);
 
-  // Handle language preferences
-
   addLanguage(patient, answers['languages-spoken']?.valueCoding);
   addLanguage(patient, answers['preferred-language']?.valueCoding, true);
 
+  // Create the patient resource
+
+  patient = await medplum.createResource(patient);
+
+  // NOTE: Updating the questionnaire response does not trigger a loop because the bot subscription
+  // is configured for "create"-only event.
+  response.subject = createReference(patient);
+  await medplum.updateResource(response);
```

Comment:
This change allows us to keep rendering the "View Intake Form" on the patient detail page.